### ISSUE TITLE
Weird text focus issue after clicking a reusable block

### ISSUE THREAD

**Issue by ZebulanStanphill** [WordPress/gutenberg#18977]
## The issue
After clicking a reusable block and then clicking a text block and then clicking another text block, you will not be able to set your text cursor in the text block. This issue is not present in Gutenberg 7.0.0.

## How to reproduce
1. Create and save a post containing a reusable block (containing whatever, it doesn't matter), followed by a Paragraph (A) and then another Paragraph (B).
2. Refresh the page.
3. Click the reusable block.
4. Click Paragraph A.
5. Click Paragraph B.
6. Try to edit the text in Paragraph B. Notice that your text cursor can't be put into Paragraph B.

## Environment
- WordPress 5.4-alpha-46814
- Gutenberg `master` as of https://github.com/WordPress/gutenberg/commit/72d9bd068716a4a1a73ccdb5d00605219d973908
- Using Dissenter browser 0.70 (based on Chromium 78)

**Comment by bobbingwide** [comment]
I've been having this problem as well. The problem doesn't appear to be limited to reusable blocks. 

### Workaround
Switch to Code editor and back again.

Is this console message related to this issue?
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in t (created by Context.Consumer)
    in oe (created by Hc)
    in div (created by Rs)
    in Rs (created by Ls)
    in Ls (created by Ks)
    in Ks (created by Hc)
    in div (created by Hc)
    in Hc (created by Vc)
    in te (created by Context.Consumer)
    in ne (created by BlockFormatControlsFill)
    in BlockFormatControlsFill (created by Context.Consumer)
    in IfBlockEditSelected(BlockFormatControlsFill) (created by Vc)
    in Vc (created by t)
```

**Comment by ZebulanStanphill** [comment]
@bobbingwide I'm not seeing any React errors in the JavaScript console, though maybe it's because I'm testing on the plugin (version 7.2.0) rather than `master` and I'm not using React DevTools or something like that.

**Comment by talldan** [comment]
This seems to have been resolved by #19839. 🎉
